### PR TITLE
fix(chat): multiple files upload

### PR DESCRIPTION
**Pull request by vimercati-samir**
What this PR does 📖
posibility to upload a bulk of files

Which issue(s) this PR fixes 🔨
[AP-788](https://satellite-im.atlassian.net/browse/AP-788)

Special notes for reviewers 🗒️

Additional comments 🎤
This branch pr should replace the one already opened for the same issue (#1676), due to rebase complications creating a new branch was the only solution

### CODE REVIEW

**Review by josephmcg — CHANGES_REQUESTED** [review]
actions.ts has a lot of unrelated updates that would break things

**Comment by molimauro** [comment]
#1676

**Comment by molimauro** [comment]
Yeah a lot of files are totally corrupted, you should not copy and paste the whole file otherwise you'll break a lot of existing functionalities. You should go line by line, let me know if you need help

**Review by molimauro — APPROVED** [review]

**Comment by molimauro** [comment]
Much better 👌

**Comment by stavares843** [comment]
Captura de ecrã 2022-03-15, às 16 34 39

it sends, but takes quite a while on that view

i'm gonna merge considering is better compared to dev

**Review by stavares843 — CHANGES_REQUESTED** [review]
i noticed that .heic preview is broken on this branch

this branch

Captura de ecrã 2022-03-15, às 16 40 19

vs dev

Captura de ecrã 2022-03-15, às 16 40 24

flow
- go to chat
- send heic file

**Comment by vimercati-samir** [comment]
> fix: incorrect merge and missing latest dev changes

should be fixed in commit 7fb078486a8b593d19217d852b30feb2a10d5ddb

**Comment by vimercati-samir** [comment]
> i noticed that .heic preview is broken on this branch
> 
> this branch
> 
> Captura de ecrã 2022-03-15, às 16 40 19
> 
> vs dev
> 
> Captura de ecrã 2022-03-15, às 16 40 24
> 
> flow
> 
> * go to chat
> * send heic file

should be fixed in commit 7fb078486a8b593d19217d852b30feb2a10d5ddb

**Comment by stavares843** [comment]
missing fixing conflict 🔨 cc @vimercati-samir

**Review by molimauro — APPROVED** [review]

**Comment by stavares843** [comment]
both @phillsatellite and CI are having this issue



https://wetransfer.com/downloads/be003b20c99e397c29e7ed8c3fc7bade20220316172325/3b5d664b7d93e2a566c4991811ae170920220316172354/5595ed

video above because is more than 100mb and github limit file size is max 100mb

dev
- send message, message appears on front of you

this branch
- send message, message doesn't appear in front of you

**Comment by vimercati-samir** [comment]
> both @phillsatellite and CI are having this issue
> 
> https://wetransfer.com/downloads/be003b20c99e397c29e7ed8c3fc7bade20220316172325/3b5d664b7d93e2a566c4991811ae170920220316172354/5595ed
> 
> video above because is more than 100mb and github limit file size is max 100mb
> 
> dev
> 
> * send message, message appears on front of you
> 
> this branch
> 
> * send message, message doesn't appear in front of you

This problem should be resolved in commit ddf370b084a77bcca665e2320862b7edfec5b720

**Review by phillsatellite — APPROVED** [review]

**Comment by stavares843** [comment]
https://user-images.githubusercontent.com/29093946/158909863-cef0f16e-91e3-4951-acf9-e946072b5fe9.mov

the scroll is fixed but is not as fluid on dev

also, there's a glitch when there are glyphs on the chat and that doesn't happen on dev

https://user-images.githubusercontent.com/29093946/158909907-3e9c2de6-1a0f-4f87-bb38-7e4b5e6c7a27.mov

**Comment by vimercati-samir** [comment]
Reverted changes on scroll in commit 1d7091cfef292393a9c11ef09734b4e37cdb7216 to close this PR and open a new one just to fix the scroll problem

**Review by stavares843 — CHANGES_REQUESTED** [review]
this issue is happening again

https://github.com/Satellite-im/Core-PWA/pull/2326#issuecomment-1069382340


new video

https://user-images.githubusercontent.com/29093946/159351824-5e443e07-f2d9-4a45-998a-509884c9d720.mov

doesn't happen on dev

**Comment by vimercati-samir** [comment]
> this issue is happening again
> 
> [#2326 (comment)](https://github.com/Satellite-im/Core-PWA/pull/2326#issuecomment-1069382340)
> 
> new video
> 
>  issue_001.mov 
> doesn't happen on dev

Should be fixed in commit a343228c6d02357be2d962d4d550aaaa25b7e753

**Comment by phillsatellite** [comment]
Side Note* when tested I found a bug that if user rapid clicks send it will only send the first image as many times as you hit send, when tested on dev the same error codes appeared and it only sent the first image, I made a ticket for it and figured I should mention that on here still 

https://satellite-im.atlassian.net/jira/software/c/projects/AP/boards/5/backlog?view=detail&selectedIssue=AP-1143&issueLimit=100

File: components/ui/Chat/Scroll/Scroll.vue [hunk 1]
```diff
@@ -102,13 +89,13 @@ export default Vue.extend({
...
       ) {
         this.$store.dispatch('ui/setIsScrollOver', true)
         return
       }
 
-      if (this.$store.state.ui.isScrollOver) {
+      if (this.ui.isScrollOver) {
         this.$store.dispatch('ui/setIsScrollOver', false)
       }
     },
```
Comment (stavares843) [inline components/ui/Chat/Scroll/Scroll.vue:101]: are these modifications on the Scroll.vue in purpose considering this PR is only to fix the multiple uploaded files?

**Comment by stavares843** [comment]
dev

Captura de ecrã 2022-03-22, às 23 13 03

this branch

Captura de ecrã 2022-03-22, às 23 13 08

File: components/ui/Chat/Scroll/Scroll.vue (hunk 1 above)
Comment (WanderingHogan) [reply components/ui/Chat/Scroll/Scroll.vue:101]: this wont affect it - when he added `...mapState(['ui', 'textile']),` he could remote $store.state

Comment (stavares843) [reply components/ui/Chat/Scroll/Scroll.vue:101]: 🔨

**Comment by stavares843** [comment]
fixed after rebasing

**Comment by WanderingHogan** [comment]
this works awesome for me, good work @vimercati-samir

**Review by WanderingHogan — APPROVED** [review]

**Comment by phillsatellite** [comment]
Noted small bug on this branch went testing, Emoji's box will get cut off if browser window is too small

Dev
Screen Shot 2022-03-23 at 10 16 34 AM


This Branch 
Screen Shot 2022-03-23 at 10 14 08 AM

**Comment by WanderingHogan** [comment]
lets skip this and open a new ticket for it

**Review by stavares843 — APPROVED** [review]

**Pull request merged**